compiler/semfold.nim

```diff
@@ -501,7 +501,7 @@ proc foldConv(n, a: PNode; g: ModuleGraph; check = false): PNode =
       rangeCheck(n, result.intVal, g)
   of tyFloat..tyFloat64:
     case srcTyp.kind
-    of tyInt..tyInt64, tyEnum, tyBool, tyChar:
+    of tyInt..tyInt64, tyUint..tyUInt64, tyEnum, tyBool, tyChar:
```

Comment:
For ``tyUInt`` and ``tyUInt64`` these values cannot be accessed via ``getOrdValue`` and so we need more special cases for these here, sorry. :-)